### PR TITLE
Adding GetRecommendedRdsInstanceType to RDS module

### PR DESCRIPTION
**Pull request by alpacamybags118**
PR to address #783 

This PR adds functionality to the RDS module to get a recommended instance type based on region, database engine, and a given list of instances. It will pick the first available one from the list or throw an error if none exist.

It uses the AWS SDK function DescribeOrderableDBInstanceOptions to achieve this.

I modeled it a lot off the EC2 get recommended instance type functionality, but it is somewhat simplified since the underlying AWS method used has less options.

### CODE REVIEW

**Comment by alpacamybags118** [comment]
Something that DescribeOrderableDBInstanceOptions exposes that I didn't think to add as an option is the DB engine-version. I believe there are cases where the version could make a difference on availability of an instance type, so i'll go ahead and add that. This code should still be reviewable as is, and I can make any feedback changes while I add that parameter.

**Comment by alpacamybags118** [comment]
Updated the code to also add engine version to the check

File: modules/aws/rds.go [hunk 1]
```diff
@@ -217,6 +218,65 @@ func NewRdsClientE(t testing.TestingT, region string) (*rds.RDS, error) {
...
+// If none of the instances provided are avaiable for your combination of region and database engine, this function will exit with an error.
+func GetRecommendedRdsInstanceType(t testing.TestingT, region string, engine string, engineVersion string, instanceTypeOptions []string) string {
+	out, err := GetRecommendedRdsInstanceTypeE(t, region, engine, engineVersion, instanceTypeOptions)
+	require.NoError(t, err)
+	return out
+}
+
+// GetRecommendedRdsInstanceTypeE takes in a list of RDS instance types (e.g., "db.t2.micro", "db.t3.micro") and returns the
+// first instance type in the list that is available in the given region and for the given database engine type.
+// If none of the instances provided are avaiable for your combination of region and database engine, this function will exit with an error.
```
Comment (brikis98) [inline modules/aws/rds.go:232]:
```suggestion
// If none of the instances provided are avaiable for your combination of region and database engine, this function will return an error.
```

File: modules/aws/rds.go [hunk 2]
```diff
@@ -217,6 +218,65 @@ func NewRdsClientE(t testing.TestingT, region string) (*rds.RDS, error) {
...
+	client, err := NewRdsClientE(t, region)
+	if err != nil {
+		return "", err
+	}
+	return GetRecommendedRdsInstanceTypeWithClientE(t, client, engine, engineVersion, instanceTypeOptions)
+}
+
+// GetRecommendedRdsInstanceTypeWithClientE takes in a list of RDS instance types (e.g., "db.t2.micro", "db.t3.micro") and returns the
+// first instance type in the list that is available in the given region and for the given database engine type.
+// If none of the instances provided are avaiable for your combination of region and database engine, this function will exit with an error.
```
Comment (brikis98) [inline modules/aws/rds.go:243]:
```suggestion
// If none of the instances provided are avaiable for your combination of region and database engine, this function will return an error.
```

**Review by brikis98 — COMMENTED** [review]
This is great, thank you! Just a couple minor NITs in the comments, and I can kick off tests.

**Comment by alpacamybags118** [comment]
Done!

**Comment by brikis98** [comment]
Thanks! Kicking off tests now.

**Comment by brikis98** [comment]
Ha! Got a test failure related to the instance type for EC2 instances:

```
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z logger.go:66: [1m[31mError: [0m[0m[1mError launching source instance: Unsupported: The requested configuration is currently not supported. Please check the documentation for supported configurations.
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z logger.go:66: 	status code: 400, request id: 4e020e26-7c45-4ce8-a161-290cd71622d1[0m
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z logger.go:66: 
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z logger.go:66: [0m  on main.tf line 26, in resource "aws_instance" "example_public":
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z logger.go:66:   26: resource "aws_instance" "example_public" [4m{[0m
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z logger.go:66: [0m
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z logger.go:66: [0m[0m
TestTerraformRemoteExecExample 2021-03-11T10:22:13Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; [31m
```

I _think_ this is exactly the issue you're fixing for RDS here, but we need to use the `GetRecommendedInstanceType` for EC2 to solve it. Any chance you'd want to update that example/test accordingly?

The other test failure is unrelated; our GCP test account is hitting some quota issues that we're cleaning up separately.

**Comment by alpacamybags118** [comment]
Yup I can update that test

**Comment by alpacamybags118** [comment]
updated!

**Comment by brikis98** [comment]
Thank you! 

One note: I think you may have accidentally checked in a `.tfvars` file in the most recent commit. Could you remove that?

**Comment by alpacamybags118** [comment]
Whoops, removed!

**Review by brikis98 — APPROVED** [review]
OK, great, thank you. I'll kick off tests again.

**Comment by brikis98** [comment]
Tests passed. Merging now.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.32.11

**Pull request merged**